driver-mapping/src/main/java/com/datastax/driver/mapping/PropertyMapper.java

```diff
@@ -144,20 +170,26 @@ private String inferColumnName() {
         if (isComputed()) {
             return annotation(Computed.class).value();
         }
-        boolean caseSensitive = false;
-        String columnName = propertyName;
+        String columnName = mapperConfiguration.getNamingStrategy().toCassandra(propertyName);
         if (hasAnnotation(Column.class)) {
             Column column = annotation(Column.class);
-            caseSensitive = column.caseSensitive();
-            if (!column.name().isEmpty())
+            if (!column.name().isEmpty()) {
                 columnName = column.name();
-        } else if (hasAnnotation(com.datastax.driver.mapping.annotations.Field.class)) {
+            }
+            else if (column.caseSensitive()) {
+                columnName = propertyName;
+            }
+        }
+        else if (hasAnnotation(com.datastax.driver.mapping.annotations.Field.class)) {
             com.datastax.driver.mapping.annotations.Field udtField = annotation(com.datastax.driver.mapping.annotations.Field.class);
-            caseSensitive = udtField.caseSensitive();
-            if (!udtField.name().isEmpty())
+            if (!udtField.name().isEmpty()) {
                 columnName = udtField.name();
+            }
+            else if (udtField.caseSensitive()) {
+                columnName = propertyName;
+            }
         }
-        return caseSensitive ? Metadata.quote(columnName) : columnName.toLowerCase();
+        return columnName.matches(UNQUOTED_COLUMN_NAME_REGEXP) ? columnName : Metadata.quote(columnName);
```

Comment:
I initially tried to as i wrote in my first comment, but it broke some other places (i can look them up although i can't recall right now) that were expecting lower cases not to be quoted.